### ISSUE TITLE
♻️ Bump `ts-jest` from 29.2.3 to latest

### ISSUE THREAD

**Issue by vd1992** [GCTC-NTGC/gc-digital-talent#11364]
## ♻️ Debt/Refactor

For some reason this bump breaks

`src/lang/lang.test.ts`

## 🕵️ Details

PR https://github.com/GCTC-NTGC/gc-digital-talent/pull/11360

Test

![image](https://github.com/user-attachments/assets/237df673-b24a-4595-9420-b713e346a007)

## ✅ Acceptance Criteria

- [ ] Dependency updated
- [ ] Tests pass

**Comment by mnigh** [comment]
Seems like it was something in 29.2.4: https://github.com/GCTC-NTGC/gc-digital-talent/pull/11175.

**Comment by esizer** [comment]
Seems we need to wait on this:

- https://github.com/kulshekhar/ts-jest/issues/4198

**Comment by mnigh** [comment]
> Seems we need to wait on this:
> 
> * [Support `Node16/NodeNext` value for `moduleResolution` kulshekhar/ts-jest#4198](https://github.com/kulshekhar/ts-jest/issues/4198)

@esizer That issue is from a year and a half ago. If we replaced jest with vitest, this issue could be closed. Any reason not to replace jest with vitest?

**Comment by esizer** [comment]
> > Seems we need to wait on this:
> > 
> > * [Support `Node16/NodeNext` value for `moduleResolution` kulshekhar/ts-jest#4198](https://github.com/kulshekhar/ts-jest/issues/4198)
> 
> [@esizer](https://github.com/esizer) That issue is from a year and a half ago. If we [replaced jest with vitest](https://vitest.dev/guide/migration#jest), this issue could be closed. Any reason not to replace jest with vitest?

No, I don't think so! It should be pretty close to a drop-in replacement.

https://vitest.dev/guide/comparisons.html#jest
